packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTree.tsx

```diff
@@ -119,6 +133,100 @@ export function ModelsTree(props: ModelsTreeProps) {
+) {
+  return function ModelsTreeRenderer(treeProps: TreeRendererProps) {
+    const { onContextMenu, renderContextMenu } = useContextMenu({ contextMenuItems: restProps.contextMenuItems });
+    const { onClearFilterClick, onFilterClick, filterDialog, containerRef } = useFilterablePresentationTree<HTMLDivElement>({ nodeLoader });
+    const isEnlarged = restProps.density === "enlarged";
+    const className = classNames("tree-widget-tree-nodes-list", { ["enlarge"]: isEnlarged });
+
+    if (isEnlarged) {
+      treeProps.nodeHeight = () => 43;
+    }
+
+    return (
+      <div className={className} ref={containerRef}>
+        <TreeRenderer
+          {...restProps}
+          {...treeProps}
+          nodeRenderer={createModelsTreeNodeRenderer(
+            { ...nodeRendererProps, isEnlarged },
+            {
+              onClearFilterClick,
+              onFilterClick,
+              onContextMenu,
+            },
+          )}
+        />
+        {renderContextMenu()}
+        {filterDialog}
+      </div>
+    );
+  };
+}
+
+interface ModelsTreeNodeActions {
+  onClearFilterClick: (nodeId: string) => void;
+
+  onFilterClick: (nodeId: string) => void;
+
+  onContextMenu: (e: React.MouseEvent, node: TreeModelNode) => void;
+}
+
+interface ModelsTreeNodeRendererProps {
+  iconsEnabled: boolean;
+
+  descriptionEnabled: boolean;
+
+  levelOffset: number;
+
+  disableRootNodeCollapse: boolean;
+
+  isEnlarged?: boolean;
+}
+
+/**
+ * Creates node renderer which renders node with eye checkbox.
+ * @public
+ */
+function createModelsTreeNodeRenderer(
+  { levelOffset, disableRootNodeCollapse, descriptionEnabled, isEnlarged }: ModelsTreeNodeRendererProps,
+  { onClearFilterClick, onFilterClick, onContextMenu }: ModelsTreeNodeActions,
+) {
+  return function ModelsTreeNodeRenderer(treeNodeProps: TreeNodeRendererProps) {
+    const expansionToggleWidth = isEnlarged ? EXPANSION_TOGGLE_WIDTH * 2 : EXPANSION_TOGGLE_WIDTH;
+    const nodeOffset = treeNodeProps.node.depth * levelOffset + (treeNodeProps.node.numChildren === 0 ? expansionToggleWidth : 0);
+
+    return (
+      <PresentationTreeNodeRenderer
+        {...treeNodeProps}
+        onContextMenu={onContextMenu}
+        onFilterClick={onFilterClick}
+        onClearFilterClick={onClearFilterClick}
+        checkboxRenderer={(checkboxProps: NodeCheckboxRenderProps) => (
+          <div className="visibility-tree-checkbox-container" style={{ marginRight: `${nodeOffset}px` }}>
+            <VisibilityTreeNodeCheckbox {...checkboxProps} />
+          </div>
+        )}
+        descriptionEnabled={descriptionEnabled}
+        imageLoader={imageLoader}
+        className={classNames(
+          "with-checkbox",
+          (treeNodeProps.node.numChildren === 0 || (disableRootNodeCollapse && treeNodeProps.node.parentId === undefined)) && "disable-expander",
+        )}
+        node={{ ...treeNodeProps.node, depth: 0 }}
```

Comment:
I think we don't need to override `depth` now that margin is disabled through CSS

Reply:
It is still needed as it adds a margin to the whole content of the node if not set.